internal/cmd/egctl/config.go

```diff
@@ -66,21 +70,43 @@ func retrieveConfigDump(args []string, includeEds bool) (*adminv3.ConfigDump, er
 	return configDump, nil
 }
 
-func portForwarder(nn types.NamespacedName) (kube.PortForwarder, error) {
+func portForwarder(nn types.NamespacedName, labelSelectors []string) (kube.PortForwarder, error) {
+	var err error
 	c, err := kube.NewCLIClient(options.DefaultConfigFlags.ToRawKubeConfigLoader())
 	if err != nil {
 		return nil, fmt.Errorf("build CLI client fail: %w", err)
 	}
 
-	pod, err := c.Pod(nn)
-	if err != nil {
-		return nil, fmt.Errorf("get pod %s fail: %w", nn, err)
+	var pod *corev1.Pod
+	if len(labelSelectors) > 0 {
+		podList, err := c.PodsForSelector(nn.Namespace, labelSelectors...)
+		if err != nil {
+			return nil, fmt.Errorf("get pod %s fail: %w", nn, err)
+		}
+
+		if len(podList.Items) == 0 {
+			return nil, fmt.Errorf("no Pods found for label selectors %+v", labelSelectors)
+		}
+		if len(podList.Items) > 1 {
```

Comment:
curious why the 1 pod limitation is added ?

Reply:
ah got it, yah a map works, thats what we recently did with `translate`
https://github.com/envoyproxy/gateway/blob/f998eacfaf54e1bfc18bb93db2074121e686ae90/internal/cmd/egctl/testdata/translate/out/from-gateway-api-to-xds.all.yaml#L2
could be done here or lets raise a issue to this can be tracked 
cc @zirain

Reply:
wouldn't `<namespace>-<name>` be tricky to interpret given that both of them can have `-` in their names?

Reply:
I think we need some spec here, maybe something like this?
```
{
    ns-1: {
        pod-name-1: {....},
        pod-name-2: {....},
    },
}
```

Reply:
we are using `ns-name` across the project as a `key` for config e.g https://github.com/envoyproxy/gateway/blob/f998eacfaf54e1bfc18bb93db2074121e686ae90/internal/gatewayapi/helpers.go#L388
```
 ns-1: {
        pod-name-1: {....},
```
gets tricky because it the definition more Kubernetesy which we should ideally avoid, if we want to support other environments

Reply:
thinking again, something like
```
 ns-1: {
        pod-name-1: {....},
```
is more readable for the end user, which might make sense

Reply:
> thinking again, something like
> 
> ```
>  ns-1: {
>         pod-name-1: {....},
> ```
> 
> is more readable for the end user, which might make sense

+1 for this style for visibility

Reply:
> thinking again, something like
> 
> ```
>  ns-1: {
>         pod-name-1: {....},
> ```
> 
> is more readable for the end user, which might make sense

+1 for this

Reply:
nice. will output in this format then, and update the PR